File: ProgramingApp/app/src/main/java/com/example/programingapp/model/Quote.java
package com.example.programingapp.model;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity
public class Quote {

    @PrimaryKey
    @NonNull
    private String id;
    private String author;
    private String en;
    private double rating;
    private int numberOfVotes;

    public Quote() {}

    @Ignore
    public Quote(String author, String en, double rating, String id, int numberOfVotes) {
        this.author = author;
        this.en = en;
        this.rating = rating;
        this.id = id;
        this.numberOfVotes = numberOfVotes;
    }

    public int getNumberOfVotes() {
        return numberOfVotes;
    }

    public void setNumberOfVotes(int numberOfVotes) {
        this.numberOfVotes = numberOfVotes;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getEn() {
        return en;
    }

    public void setEn(String en) {
        this.en = en;
    }

    public double getRating() {
        return rating;
    }

    public void setRating(double rating) {
        this.rating = rating;
    }
}

File: ProgramingApp/app/src/main/java/com/example/programingapp/model/Quote.kt
package com.example.programingapp.model

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity
class Quote constructor(@PrimaryKey val id: String) {

    var author: String? = null
    var en: String? = null
    var rating: Double = 0.toDouble()
    var numberOfVotes: Int = 0

    constructor(author: String, en: String, rating: Double, id: String, numberOfVotes: Int) : this(id) {
        this.author = author
        this.en = en
        this.rating = rating
        this.numberOfVotes = numberOfVotes
    }
}
